### PR TITLE
PRM-1939 Updated 1811 mapping/model for new API Spec

### PR DESCRIPTION
**Pull request by AndrewRButcher**

### CODE REVIEW

File: app/config/AppConfig.scala [hunk 1]
```diff
@@ -26,15 +26,19 @@ import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
...
 
   def queryParametersForGetFinancialDetail(dateFrom: LocalDate, dateTo: LocalDate): String = {
-    s"?dateFrom=$dateFrom&dateTo=$dateTo&onlyOpenItems=${config.get[Boolean]("eis.onlyOpenItems")}" +
-      s"&includeStatistical=${config.get[Boolean]("eis.includeStatistical")}" +
-      s"&includeLocks=${config.get[Boolean]("eis.includeLocks")}" +
-      s"&calculateAccruedInterest=${config.get[Boolean]("eis.calculateAccruedInterest")}" +
-      s"&removePOA=${config.get[Boolean]("eis.removePOA")}" +
-      s"&customerPaymentInformation=${config.get[Boolean]("eis.customerPaymentInformation")}"
+    s"?dateFrom=$dateFrom&dateTo=$dateTo" +
+      s"&includeClearedItems=${!config.get[Boolean](" eis.includeCleared")}" +
```
Comment (LewisMitchell) [inline app/config/AppConfig.scala:33]: Why is this not?
```suggestion
      s"&includeClearedItems=${config.get[Boolean]("eis.includeCleared")}" +
```

File: app/services/PenaltiesFrontendService.scala [hunk 2]
```diff
@@ -16,29 +16,28 @@
...
       _.details.map(_.map(
         oldLPPDetails => {
           val isAdditional = oldLPPDetails.penaltyCategory.equals(LPPPenaltyCategoryEnum.SecondPenalty)
-          val firstAndMaybeSecondPenalty = financialDetails.financialDetails.filter(_.chargeReference.exists(_.equals(oldLPPDetails.principalChargeReference)))
+          val firstAndMaybeSecondPenalty = financialDetails.documentDetails.get.filter(_.chargeReferenceNumber.exists(_.equals(oldLPPDetails.principalChargeReference)))
           val penaltyToCopy = firstAndMaybeSecondPenalty.find(lpp => {
-              if(isAdditional) MainTransactionEnum.secondCharges.contains(lpp.mainTransaction.get)
-              else MainTransactionEnum.firstCharges.contains(lpp.mainTransaction.get)
+              if(isAdditional) MainTransactionEnum.secondCharges.contains(lpp.lineItemDetails.get.head.mainTransaction.get)
+              else MainTransactionEnum.firstCharges.contains(lpp.lineItemDetails.get.head.mainTransaction.get)
```
Comment (LewisMitchell) [inline app/services/PenaltiesFrontendService.scala:34]: Adding this for traceability, we are going to ask ETMP to see if LPP1 and LPP2 come under different charges

File: it/services/GetFinancialDetailsServiceISpec.scala [hunk 3]
```diff
@@ -30,113 +30,38 @@ class GetFinancialDetailsServiceISpec extends IntegrationSpecCommonBase with ETM
...
-          contractObjectType = Some("ABCD"),
-          contractObject = Some("00000003000000002757"),
-          sapDocumentNumber = Some("1040000872"),
-          sapDocumentNumberItem = Some("XM00"),
-          subTransaction = Some("5678"),
-          clearedAmount = Some(123.45),
-          accruedInterest = Some(543.21)
-        )
+
+    val getFinancialDetailsModel: FinancialDetails = getFinancialDetails.FinancialDetails(
```
Comment (LewisMitchell) [inline it/services/GetFinancialDetailsServiceISpec.scala:34]: Do you need the getFinancialDetails prefixes?

File: test/connectors/getFinancialDetails/GetFinancialDetailsConnectorSpec.scala [hunk 4]
```diff
@@ -52,86 +53,12 @@ class GetFinancialDetailsConnectorSpec extends SpecBase with LogCapturing {
...
-        contractObjectType = None,
-        contractObject = None,
-        sapDocumentNumber = None,
-        sapDocumentNumberItem = None,
-        subTransaction = None,
-        clearedAmount = None,
-        accruedInterest = None
-      )
+
+  val mockGetFinancialDetailsModelAPI1811: FinancialDetails = getFinancialDetails.FinancialDetails(
```
Comment (LewisMitchell) [inline test/connectors/getFinancialDetails/GetFinancialDetailsConnectorSpec.scala:57]: Do you need the getFinancialDetails prefix?

File: test/connectors/parsers/GetFinancialDetailsParserSpec.scala [hunk 5]
```diff
@@ -28,89 +29,13 @@ import uk.gov.hmrc.http.HttpResponse
...
-        contractAccount = None,
-        contractObjectType = None,
-        contractObject = None,
-        sapDocumentNumber = None,
-        sapDocumentNumberItem = None,
-        subTransaction = None,
-        clearedAmount = None,
-        accruedInterest = None
-      )
+  val mockGetFinancialDetailsModelAPI1811: FinancialDetails = getFinancialDetails.FinancialDetails(
```
Comment (LewisMitchell) [inline test/connectors/parsers/GetFinancialDetailsParserSpec.scala:34]: Do you need the getFinancialDetails prefix?

File: test/connectors/parsers/GetFinancialDetailsParserSpec.scala [hunk 6]
```diff
@@ -148,11 +73,6 @@ class GetFinancialDetailsParserSpec extends AnyWordSpec with Matchers with LogCa
         result.isRight shouldBe true
         result.right.get.asInstanceOf[GetFinancialDetailsSuccessResponse].financialDetails shouldBe mockGetFinancialDetailsModelAPI1811
       }
-
-      s"the body is malformed - returning a $Left $GetFinancialDetailsMalformed" in {
-        val result = GetFinancialDetailsParser.GetFinancialDetailsReads.read("GET", "/", mockOKHttpResponseWithInvalidBody)
-        result.isLeft shouldBe true
-      }
```
Comment (LewisMitchell) [inline test/connectors/parsers/GetFinancialDetailsParserSpec.scala:155]: Could we add this test back in?

File: test/controllers/PenaltiesFrontendControllerSpec.scala [hunk 7]
```diff
@@ -304,65 +304,14 @@ class PenaltiesFrontendControllerSpec extends SpecBase with LogCapturing {
...
-              sapDocumentNumber = Some("1"),
-              sapDocumentNumberItem = Some("1"),
-              subTransaction = Some("1"),
-              clearedAmount = Some(123.45),
-              accruedInterest = Some(123.45)
-            )
-          )
-        )
+      val financialDetails: FinancialDetails = getFinancialDetails.FinancialDetails(
+        documentDetails = Some(Seq(getFinancialDetails.DocumentDetails(
```
Comment (LewisMitchell) [inline test/controllers/PenaltiesFrontendControllerSpec.scala:308]: Do you need the getFinancialDetails prefix?

File: test/services/GetFinancialDetailsServiceSpec.scala [hunk 8]
```diff
@@ -46,87 +46,11 @@ class GetFinancialDetailsServiceSpec extends SpecBase with FeatureSwitching {
...
-          contractAccount = Some("X"),
-          contractObjectType = Some("ABCD"),
-          contractObject = Some("00000003000000002757"),
-          sapDocumentNumber = Some("1040000872"),
-          sapDocumentNumberItem = Some("XM00"),
-          subTransaction = Some("5678"),
-          clearedAmount = Some(123.45),
-          accruedInterest = Some(543.21)
-        )
+    val mockGetFinancialDetailsResponseAsModel: FinancialDetails = getFinancialDetails.FinancialDetails(
```
Comment (LewisMitchell) [inline test/services/GetFinancialDetailsServiceSpec.scala:49]: Do you need the getFinancialDetails prefix?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/config/AppConfig.scala (hunk 1 above)
Comment (AndrewRButcher) [reply app/config/AppConfig.scala]: Carry over from initially using old config values, have rectified

File: it/services/GetFinancialDetailsServiceISpec.scala (hunk 3 above)
Comment (AndrewRButcher) [reply it/services/GetFinancialDetailsServiceISpec.scala]: Removed ✅

File: test/connectors/getFinancialDetails/GetFinancialDetailsConnectorSpec.scala (hunk 4 above)
Comment (AndrewRButcher) [reply test/connectors/getFinancialDetails/GetFinancialDetailsConnectorSpec.scala]: Removed ✅

File: test/connectors/parsers/GetFinancialDetailsParserSpec.scala (hunk 5 above)
Comment (AndrewRButcher) [reply test/connectors/parsers/GetFinancialDetailsParserSpec.scala]: Removed ✅

File: test/connectors/parsers/GetFinancialDetailsParserSpec.scala (hunk 6 above)
Comment (AndrewRButcher) [reply test/connectors/parsers/GetFinancialDetailsParserSpec.scala:155]: Added back in ✅

File: test/controllers/PenaltiesFrontendControllerSpec.scala (hunk 7 above)
Comment (AndrewRButcher) [reply test/controllers/PenaltiesFrontendControllerSpec.scala:308]: Removed ✅

File: test/services/GetFinancialDetailsServiceSpec.scala (hunk 8 above)
Comment (AndrewRButcher) [reply test/services/GetFinancialDetailsServiceSpec.scala]: Removed ✅

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**